README.md

```diff
@@ -360,12 +360,104 @@ As long as the string you pass to the constructor is a valid selector string the
 
 *Note: at this time there is no support for multiple HTMLElements as parameters. If you are passing an HTMLElement as the parameter rather than a selector string you cannot pass multiple controls. If you do wish to use multiple controls, pass the plugin a selector string instead.
 
+## Authoring Plugins
+
+Writing a plugin for Container is relatively simple but there are some differences based on whether it is going to be a built-in plugin, or external.
+
+Plugins for Container should take the form of an ES6 module, preferably with a named export. E.g.:
+```javascript
+export class ContainerPlugin() {...}
+```
+
+When developing a plugin for Container v2.x there are 3 main lifecycle hooks to keep in mind: `preload`, `init`, and `start`.
+
+`preload()`: Preload expects an async method or at least a returned promise. The method is passed the entire [PluginManager](https://github.com/SpringRoll/SpringRollContainer/blob/main/src/PluginManager.js) object. In general all plugins need to access is the [Bellhop client](https://github.com/SpringRoll/Bellhop), but the entire object is available should you require it.
+Note: if this method is missing or fails to resolve, Container will automatically discard your plugin and it will not be loaded in properly.
+
+Example:
+```javascript
+export class ContainerPlugin() {
+  constructor() {...}
+
+  async preload({ client }) {
+    this.client = client;
+  }
+}
+```
+
+`init()`: the init method is called once the preload promises have all resolved. This method would be where you do most of your plugins setup that involves interacting with the Springroll Application. Any event listeners (bellhop or otherwise), or additional data fetching should go here. Similarly to `preload()` init is also passed the `PluginManager` class object if required.
+
+`start()`: Generally all that your plugin will require is an `init()` method. But, if the case arises where your plugin depends on another plugin the start method exists. This is called after all preloaded plugins have finished their init calls and serves a similar purpose. This can help ease race conditions between plugins, as they are not guarenteed to call `init` in a consistent order. Similarly to `preload()` init is also passed the `PluginManager` class object if required.
```

Comment:
```suggestion
`start()`: The `start()` method is called after all preloaded plugins have finished their `init()` calls. This method allows plugins to depend on others by allowing a second plugin to `start()` after a first plugin has ran `init()`. This can help ease race conditions between plugins, as they are not guarenteed to call `init()` in a consistent order. Similarly to `preload()`, `init()` is also passed the `PluginManager` class object if required.
```